common/include/dummy_protocol.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * dummy_protocol.h
+ *
+ *  Created on: 2020��10��17��
+ *      Author: duanning
+ */
+
+#ifndef DUMMY_PROTOCOL_H_
+#define DUMMY_PROTOCOL_H_
+
+#include <iostream>
+#include "abstract_protocol.h"
+
+class DummyProtocol: public AbstractProtocol {
+public:
+	DummyProtocol() {};
+	~DummyProtocol() {};
+
+	virtual error_condition PackFrame(\
+			const vector<uint8_t>& buffer, vector<uint8_t>& out_data) {};
+	virtual error_condition UnpackFrame(\
+			vector<uint8_t>& buffer, vector<uint8_t>& valid_data) {};
+private:
+	virtual error_condition IsFrameComplete(const vector<uint8_t>& buffer) {};
```

Comment:
These functions needs to return a bool value, and we need to always return `true`.